Tests/SentryTests/SentryCrash/SentryCrashInstallationTests.m

```diff
@@ -0,0 +1,146 @@
+#import "SentryCrash.h"
+#import "SentryCrashCachedData.h"
+#import "SentryCrashInstallation+Private.h"
+#import "SentryCrashInstallation.h"
+#import "SentryCrashMonitor.h"
+#import "SentryCrashMonitor_MachException.h"
+#import "SentryNSNotificationCenterWrapper.h"
+#import "SentryTests-Swift.h"
+#import <XCTest/XCTest.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SentryCrashTestInstallation : SentryCrashInstallation
+
+@end
+
+@implementation SentryCrashTestInstallation
+
+- (instancetype)initForTesting
+{
+    if (self = [super initWithRequiredProperties:[NSArray new]]) { }
+    return self;
+}
+
+@end
+
+@interface SentryCrashInstallationTests : XCTestCase
+
+@property (nonatomic, strong) TestNSNotificationCenterWrapper *notificationCenter;
+
+@end
+
+@implementation SentryCrashInstallationTests
+
+- (SentryCrashTestInstallation *)getSut
+{
+    SentryCrashTestInstallation *installation =
+        [[SentryCrashTestInstallation alloc] initForTesting];
+    self.notificationCenter = [[TestNSNotificationCenterWrapper alloc] init];
+    [[SentryCrash sharedInstance] setSentryNSNotificationCenterWrapper:self.notificationCenter];
+    return installation;
+}
+
+- (void)testUninstall
+{
+    SentryCrashTestInstallation *installation = [self getSut];
+
+    [installation install];
+
+    SentryCrashMonitorType monitorsAfterInstall = [SentryCrash sharedInstance].monitoring;
+
+    [installation uninstall];
+
+    [self assertUninstalled:installation monitorsAfterInstall:monitorsAfterInstall];
+}
+
+- (void)testUninstall_CallsRemoveObservers
+{
+    SentryCrashTestInstallation *installation = [self getSut];
+
+    [installation install];
+    [installation uninstall];
+
+#if SentryCrashCRASH_HAS_UIAPPLICATION
+    XCTAssertEqual(5, self.notificationCenter.removeWithNotificationInvocationsCount);
```

Comment:
This test doesn't run. You can change this to `XCTAssertEqual(5, 3);` and it still succeeds.

Reply:
Upsi, will fix that.

Reply:
https://github.com/getsentry/sentry-cocoa/pull/2500